### DISCUSSION TITLE
Tree Checkbox API

### DISCUSSION THREAD

**Discussion by isidorn** [microsoft/vscode-discussions#158] [Announcements]
The proposed `checkboxState` lets you add checkboxes to tree items.

```ts
this.checkboxState = { state: vscode.TreeItemCheckboxState.Unchecked, tooltip: 'mark file as viewed' };
```

![image](https://user-images.githubusercontent.com/1926584/194511750-95c81400-dd02-4a50-9f02-85f2058b57d2.png)

Try it out and let us know what you think about this proposed API. fyi @alexr00

**Comment by mattseddon** [comment]
Cross-posting from the original issue (https://github.com/microsoft/vscode/issues/116141#issuecomment-1270891813):

> Would it be possible to add an indeterminate (middle) state checkbox to the API? This would match the webview toolkit: https://github.com/microsoft/vscode-webview-ui-toolkit/blob/main/src/checkbox/README.md

**Comment by KrabsWong** [comment]
Will we have an expected date to finalize the API?

**Reply by isidorn** [reply]
@krabswong it usually takes us at least a couple of milestones to finalize an API. And it really helps if extension authors try out the API and provide feedback to us in the meantime. That helps speed up the finalization process.
